### PR TITLE
[Quant][Inductor] Add qlinear_pointwise.binary op for X86Inductor backend

### PR DESCRIPTION
**Pull request by Xia-Weiwen**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #123240
* #122667
* #122593
* #122387
* __->__ #123144

**Note**: This is a reopen of https://github.com/pytorch/pytorch/pull/122288, which was merged by `ghstack land` to its base (not main) by mistake.

**Description**
Add qlinear_binary op for X86Inductor backend of quantization PT2E. It only supports `add` and `add_relu` now.
It will use post op sum if the extra input has the same dtype as output. Otherwise, it uses binary add.
```
+-------------------+--------------+---------------+
| Extra input dtype | Output dtype | Post op       |
+-------------------+--------------+---------------+
| Fp32/bf16         | fp32/bf16    | sum or add*   |
+-------------------+--------------+---------------+
| Fp32/bf16         | int8         | add           |
+-------------------+--------------+---------------+
| int8              | fp32/bf16    | not supported |
+-------------------+--------------+---------------+
| int8              | int8         | sum           |
+-------------------+--------------+---------------+
*Use sum if extra input and output have the same dtype; otherwise use add.
```

**Test plan**
python test_quantization.py -k test_qlinear_add_pt2e
python test_quantization.py -k test_qlinear_add_relu_pt2e

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Comment by Xia-Weiwen** [comment]
Hi @jgong5  @leslie-fang-intel. Could you please help review this PR? This is a reopen of https://github.com/pytorch/pytorch/pull/122288, which was merged by `ghstack land` to its base (not main) by mistake and thus discarded. Thanks!

**Review by leslie-fang-intel — APPROVED** [review]

**Comment by Xia-Weiwen** [comment]
Hi @jerryzh168 Could you please review this PR? Thanks!

**Review by jgong5 — APPROVED** [review]

**Comment by Xia-Weiwen** [comment]
Hi @jerryzh168 Could you please take a look? Thanks!

File: test/quantization/core/test_quantized_op.py [hunk 1]
```diff
@@ -4331,21 +4382,41 @@ def _test_qlinear_pt2e_helper(self, post_op_to_qlinear_ref_dict, supported_post_
...
     @skipIfNoONEDNN
-    def test_qlinear_linear_relu_pt2e(self):
-        post_op_to_qlinear_ref_dict = {
-            'none': torch.ops.quantized.linear,
-            'relu': torch.ops.quantized.linear_relu,
-        }
-        supported_post_ops = ['none', 'relu']
-        post_op_algorithms = []
-        self._test_qlinear_pt2e_helper(post_op_to_qlinear_ref_dict, supported_post_ops, post_op_algorithms)
+    def test_qlinear_pt2e(self):
```
Comment (jerryzh168) [inline test/quantization/core/test_quantized_op.py:4385]: nit: as a follow up maybe you can add a new test_quantized_op_onednn.py file, so it's not confused with the existing quantized ops that is using quantized tensors.

**Review by jerryzh168 — APPROVED** [review]

File: test/quantization/core/test_quantized_op.py (hunk 1 above)
Comment (Xia-Weiwen) [reply test/quantization/core/test_quantized_op.py:4385]: Makes sense. Thanks.

**Pull request closed without merging**